jwst/saturation/saturation.py

```diff
@@ -160,6 +165,24 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
                                         irs2_mask, detector)
             # check for saturation
             flag_temp = np.where(sci_temp >= sat_thresh, SATURATED, 0)
+            if group == 2:
+                # Identify groups which we wouldn't expect to saturate by the third group,
+                # on the basis of the first group
+                mask = sci_temp[ints, 0, ...] / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                # Identify groups with suspiciously large values in the second group
+                mask &= sci_temp[ints, 1, ...] > sat_thresh / len(read_pattern[1])
+
+                # Identify groups that are saturated in the third group
+                dq_temp = x_irs2.from_irs2(groupdq[ints, 2, :, :], irs2_mask, detector)
+                gp3mask = np.where(dq_temp & SATURATED, True, False)
```

Comment:
This code won't work here; it's looking for where SATURATED is set in groupdq for group==2, but the flags aren't passed from flag_temp into groupdq until after this code block executes.  Likewise for the stcal PR.